### PR TITLE
monores refactorized

### PR DESCRIPTION
**Pull request by Vilax**
monores refactorized and localdeblur modified according to monores changes

https://github.com/I2PC/xmipp/pull/266 is a requirement.

### CODE REVIEW

File: xmipp3/protocols/protocol_resolution_monogenic_signal.py [hunk 1]
```diff
@@ -99,14 +97,15 @@ def _defineParams(self, form):
...
-#                       'See [[http://xmipp.cnb.csic.es/twiki/bin/view/Xmipp/Symmetry][Symmetry]]'
-#                       'for a description of the symmetry groups format,' 
-#                       'If no symmetry is present, give c1.')
+        form.addParam('MaskExcl', PointerParam, pointerClass='VolumeMask',
+		      expertLevel=LEVEL_ADVANCED,
+                      condition='(halfVolumes) or (not halfVolumes)',
+                      allowsNull=True,
+                      label="Exclude Area", 
+                      help='The mask determines the area of the protein to be'
+		      'excluded in the estimation of the local resolution')
```
Comment (rsanchezgarc) [inline xmipp3/protocols/protocol_resolution_monogenic_signal.py:106]: it should be clarified if the mask (I assume binary) will exclude 1 regions or 0 regions

File: xmipp3/viewers/viewer_resolution_monogenic_signal.py [hunk 2]
```diff
@@ -226,9 +231,9 @@ def createChimeraScript(self):
         fnRoot = "extra/"
         scriptFile = self.protocol._getPath('Chimera_resolution.cmd')
         fhCmd = open(scriptFile, 'w')
```
Comment (rsanchezgarc) [inline xmipp3/viewers/viewer_resolution_monogenic_signal.py:233]: could you use the with open("name") as f:  

option instead explicit open and close?

File: xmipp3/protocols/protocol_resolution_monogenic_signal.py [hunk 3]
```diff
@@ -99,14 +97,15 @@ def _defineParams(self, form):
...
-        group = form.addGroup('Extra parameters')
-#         group.addParam('symmetry', StringParam, default='c1',
-#                       label="Symmetry",
-#                       help='Symmetry group. By default = c1.'
-#                       'See [[http://xmipp.cnb.csic.es/twiki/bin/view/Xmipp/Symmetry][Symmetry]]'
-#                       'for a description of the symmetry groups format,' 
-#                       'If no symmetry is present, give c1.')
+        form.addParam('MaskExcl', PointerParam, pointerClass='VolumeMask',
+		      expertLevel=LEVEL_ADVANCED,
+                      condition='(halfVolumes) or (not halfVolumes)',
```
Comment (dmaluenda) [inline xmipp3/protocols/protocol_resolution_monogenic_signal.py:102]: this condition seems unnecessary. If something or not something is always true..

File: xmipp3/protocols/protocol_resolution_monogenic_signal.py [hunk 4]
```diff
@@ -126,48 +125,25 @@ def _defineParams(self, form):
...
                       condition = 'halfVolumes',
-                      help='Select yes if the volume present noise outside the mask.'
-                      ' Otherwise, select No.')
+                      help='(Yes recommended) the noise distribution will be estimated'
+		      ' in the protein region (inside the mask) by means of the '
+		      'difference of both half maps.')
+
+        form.addParam('gaussianNoise', BooleanParam, expertLevel=LEVEL_ADVANCED,
+                      default=False,
+                      label="Consider noise gaussian?",
```
Comment (dmaluenda) [inline xmipp3/protocols/protocol_resolution_monogenic_signal.py:138]: Minor: "Consider Gaussian noise?"

File: xmipp3/viewers/viewer_resolution_monogenic_signal.py [hunk 5]
```diff
@@ -254,74 +259,96 @@ def createChimeraScript(self):
...
+            if self.protocol.halfVolumes.get() is True:
                 smprt = self.protocol.inputVolume.get().getSamplingRate()
             else:
                 smprt = self.protocol.inputVolumes.get().getSamplingRate()
 
             imageFile = replaceExt(imageFile, 'vol')
             fhCmd.write("open %s\n" % fninput)
-            fhCmd.write("open %s\n" %  abspath(imageFile))
-            fhCmd.write("volume #0 voxelSize %s\n" % (str(smprt)))
+	    fhCmd.write("open %s\n" %  abspath(imageFile))
```
Comment (dmaluenda) [inline xmipp3/viewers/viewer_resolution_monogenic_signal.py:272]: Minor: Be careful with the indentations

**Comment by dmaluenda** [comment]
This PR should be against `python3_migration`. Hints to do it:

1. Install Scipion3 from scratch via [scipion-installer](https://pypi.org/project/scipion-installer/)
2. The installer should install also xmipp inside the scipion3 folder (scipion3/xmipp-bundle)
3. `cd scipion3/xmipp-bundle`
4. `./xmipp git checkout vi_monores_refactorized`
5. `./xmipp git checkout -b vi_monores_refactorized_3`
6. `./xmipp git merge origin/python3_migration`
7. Resolve all conflicts inside xmipp and scipion-em-xmipp repos
8. Make a new PR from `vi_monores_refactorized_3` to `python3_migration`.

[5 is optional to work in a new branch, while `vi_monores_refactorized` still there as a backup...]

Ask help if you need.

**Pull request closed without merging**